In [1]:
import joblib
import pandas as pd

In [2]:
lr_model = joblib.load('../models/customer_churn_logistic_regression.pkl')
rf_model = joblib.load('../models/customer_churn_random_forest.pkl')

In [3]:
new_customer = pd.DataFrame({
    'gender': ['Female'],
    'SeniorCitizen': [0],
    'Partner': ['Yes'],
    'Dependents': ['No'],
    'tenure': [5],
    'PhoneService': ['Yes'],
    'MultipleLines': ['No'],
    'InternetService': ['Fiber optic'],
    'OnlineSecurity': ['No'],
    'OnlineBackup': ['Yes'],
    'DeviceProtection': ['No'],
    'TechSupport': ['No'],
    'StreamingTV': ['Yes'],
    'StreamingMovies': ['Yes'],
    'Contract': ['Month-to-month'],
    'PaperlessBilling': ['Yes'],
    'PaymentMethod': ['Electronic check'],
    'MonthlyCharges': [85],
    'TotalCharges': [427]
})

In [4]:
lr_model.predict(new_customer)

array([1])

In [6]:
lr_model.predict_proba(new_customer)

array([[0.11827132, 0.88172868]])

In [7]:
rf_model.predict(new_customer)

array([1])

In [8]:
rf_model.predict_proba(new_customer)

array([[0.13349468, 0.86650532]])

## Prediction Example

A sample customer was passed through the trained pipeline to generate a churn prediction.

Both models predicted that the sample customer is likely to churn with the probability of 88% and 86%.